test/e2e/v3/plugin_cluster_test.go

```diff
@@ -398,43 +366,20 @@ func curlMetrics(kbc *utils.TestContext) string {
 	return metricsOutput
 }
 
-// TODO: Remove this when and other related functions when
-// tests have been changed to use a better implementation.
-// ServiceAccountToken creates the ServiceAccount token Secret.
-// This is to be used when Kubernetes cluster version is >= 1.24.
-// In k8s 1.24+ a ServiceAccount token Secret is no longer
-// automatically generated
-func ServiceAccountToken(kbc *utils.TestContext) {
-	// As of Kubernetes 1.24 a ServiceAccount no longer has a ServiceAccount token
-	// secret autogenerated. We have to create it manually here
+// ServiceAccountToken performs a kubectl create token to request a new Token
+// for the service account in question and use that token for performing the
+// curl command calls as required. Please keep in mind that the output of
+// kubectl create token is not base64 encoded. It is the JWT token itself
+func ServiceAccountToken(kbc *utils.TestContext) string {
 	By("Creating the ServiceAccount token")
-	secretFile, err := createServiceAccountToken(kbc.Kubectl.ServiceAccount, kbc.Dir)
-	Expect(err).NotTo(HaveOccurred())
+	var out string
+	var err error
 	Eventually(func() error {
-		_, err = kbc.Kubectl.Apply(true, "-f", secretFile)
+		out, err = kbc.Kubectl.Command(
+			"create", "token", kbc.Kubectl.ServiceAccount,
+			"--audience",
+			fmt.Sprintf("https://e2e-%s-controller-manager-metrics-service.%s.svc:8443", kbc.TestSuffix, kbc.Kubectl.Namespace))
```

Comment:
@everettraven  @camilamacedo86 Do you want these items to be more specific or more free in terms of what is in the Token ?

Reply:
Hi @harshanarayana I unable the tests with this change they are not working
It is fine do the change here directly
However, we test against 1.18 so if you change it for a syntax that does not work on 1.18 we need to have the logic implemented to do that only when is > than x. ( as it is now only for example scaffolding the token when the test is done against 1.24 only )

Btw, thx for your help on this one. 🥇

Reply:
Thanks @camilamacedo86  Let me check and update it. I am yet to get this tested locally as well.

Reply:
I've addressed the required changes. For now keeping the usage of `kubectl create token` for `v1.24+` and using the template to render a new secret that will provide the token required to perform the `curl` operation for Metrics fetch. 

> Auto-generated tokens should not be scraped for use outside pods on any version

A bit of the code has been refactored to address what @liggitt suggested.  So, all versions now use a secret of type `kubernetes.io/service-account-token` to get the token required to do the operations. PTAL. Hope this addresses the request

Reply:
Marking this as resolved. PTAL at the changes that has addressed your suggested changes. @camilamacedo86 . Thanks